frontend/src/main/scala/bloop/reporter/BspProjectReporter.scala

```diff
@@ -16,22 +16,40 @@ import scala.collection.mutable
 import scala.util.Try
 import scala.concurrent.Promise
 import bloop.CompileOutPaths
+import monix.execution.atomic.AtomicInt
 
 final class BspProjectReporter(
     val project: Project,
     override val logger: BspServerLogger,
     cwd: AbsolutePath,
     config: ReporterConfig,
     reportAllPreviousProblems: Boolean,
-    override val _problems: mutable.Buffer[ProblemPerPhase] = mutable.ArrayBuffer.empty
+    override val _problems: Reporter.Buffer[ProblemPerPhase]
 ) extends Reporter(logger, cwd, config, _problems) {
+
+  def this(
+      project: Project,
+      logger: BspServerLogger,
+      cwd: AbsolutePath,
+      config: ReporterConfig,
+      reportAllPreviousProblems: Boolean
+  ) =
+    this(
+      project,
+      logger,
+      cwd,
+      config,
+      reportAllPreviousProblems,
+      createBuffer[ProblemPerPhase](project)
+    )
+
   private lazy val taskId = logger.nextTaskId
 
   /** A cycle count, initialized to 0 when it's a no-op. */
-  private var cycleCount: Int = 0
+  private val cycleCount: AtomicInt = AtomicInt(0)
 
   /** A thread-safe map with all the files under compilation. */
-  private val compilingFiles = mutable.HashMap.empty[File, Boolean]
+  private val compilingFiles = TrieMap.empty[File, Boolean]
```

Comment:
Let's use `ConcurrentHashMap` instead of `TrieMap`

Reply:
I've used a `TrieMap` for coherence with the other fields defined in the same class (see [here](https://github.com/scalacenter/bloop/blob/master/frontend/src/main/scala/bloop/reporter/BspProjectReporter.scala#L37-L40)). But if you prefer a `ConcurrentHashMap` just for this field, I'm of course happy to make the change. Let me know!

Reply:
Yes, please, let's use `ConcurrentHashMap` here, this field is accessed quite often so I prefer to use a more efficient implementation than `TrieMap`

Reply:
You got it. Will mark this resolved.